src/dupsfindr.go

```diff
@@ -17,97 +18,95 @@ func main() {
 	directory := flag.String("directory", "", "the absolute directorypath to search for duplicates")
 	flag.Parse()
 
-	fmt.Println("DupsFindr started with: ", *directory)
+	if len(*directory) == 0 {
+		panic("no directory provided")
+	}
 
-	if len(*directory) > 0 {
-		files := make(chan string, 10)
+	fmt.Println("DupsFindr started with: ", *directory)
 
-		allFilesWithoutDuplicates := make([]string, 0, 5)
+	files := make(chan string, 10)
+	filesWithoutDuplicates := make([]string, 0, 5)
 
-		var wg sync.WaitGroup
-		wg.Add(1)
-		// read the directory and send all files to the channel
-		go readDirectory(*directory, files, &wg)
+	var wg sync.WaitGroup
+	// read the directory and send all files to the channel
+	go readDirectory(*directory, files, &wg)
 
-		fmt.Println()
-		counter := 0
-		wg.Add(1)
-		// recieve from files and add to slice
-		go readFiles(files, &allFilesWithoutDuplicates, &counter, &wg)
+	counter := 0
+	// recieve from files and add to slice
+	go readFiles(files, &filesWithoutDuplicates, &counter, &wg)
 
-		wg.Wait()
-		// is this okay, or should it be closed after the last directory was read?
-		close(files)
+	wg.Wait()
+	// is this okay, or should it be closed after the last directory was read?
+	close(files)
 
-		fmt.Println()
-		fmt.Println()
-		fmt.Println(allFilesWithoutDuplicates)
-	}
-	fmt.Println()
-	fmt.Println()
-	fmt.Println("Executed in: ", time.Since(start))
+	fmt.Printf("\n\n%v", filesWithoutDuplicates)
+	fmt.Printf("\n\nExecuted in: %v", time.Since(start))
 }
 
 func readDirectory(directory string, files chan<- string, wg *sync.WaitGroup) {
+	wg.Add(1)
```

Comment:
I moved `Add()` into the function so that a call to `readDirectory()` will always be atomic and can be used without having to call `Add()` yourself.